File: Easy/160_Intersection_of_Two_Linked_Lists/Solution.cpp
#include "../../devlibs/cpp/cpphelper.h"

using namespace std;


// NOTE:
// For LeetCode submission, copy only the `class Solution` part.

/**
 * Definition for singly-linked list.
 * struct ListNode {
 *     int val;
 *     ListNode *next;
 *     ListNode(int x) : val(x), next(NULL) {}
 * };
*/

class Solution {
public:
    ListNode *getIntersectionNode(ListNode *headA, ListNode *headB) {
        for (ListNode *tmp = headA; tmp != NULL; tmp = tmp->next) {
            for (ListNode *tmpB = headB; tmpB != NULL; tmpB = tmpB->next) {
                if (tmp == tmpB) {
                    return tmp;
                }
            }
        }

        return NULL;
    }
};


int main(int argc, char* argv[]) {
    ios::sync_with_stdio(false);
    cin.tie(NULL);

    Solution solution;

    // test cases 1
    // test cases 2

    return 0;
}

File: Easy/160_Intersection_of_Two_Linked_Lists/Solution.java





public class Solution {
    public ListNode getIntersectionNode(ListNode headA, ListNode headB) {
        for (ListNode tmp = headA; tmp != null; tmp = tmp.next) {
            for (ListNode tmpB = headB; tmpB != null; tmpB = tmpB.next) {
                if (tmp == tmpB) {
                    return tmp;
                }
            }
        }

        return null;
    }

    public static void main(String[] args) {
        Solution solution = new Solution();

        // test cases 1
        // test cases 2
        
        
    }
}


// Definition for singly-linked list.
class ListNode {
    int val;
    ListNode next;
    ListNode(int x) {
        val = x;
        next = null;
    }
}